src/apps/editor/components/matrix/form-renderer.tsx

```diff
@@ -75,31 +77,51 @@ export const FormRenderer: React.FC<Props> = ({
 
   return (
     <VCol variant="md" className={twMerge('w-full', depth > 0 && 'pl-4')}>
-      <Input
-        variant="plain"
-        defaultValue={value}
-        placeholder={type === 'array' ? 'Array key' : objectPlaceholder}
-        onBlur={onFieldUpdate(field.id, path)}
-        icon={<XIcon className="text-muted-foreground" />}
-        onIconClick={onRemoveField(field.id, path)}
-        disabled={arrayChild && type === 'object'}
-      />
-      {child.length > 0 ? (
-        <VCol variant="md" className="w-full border-l border-dashed">
-          {child.map((field) => (
-            <FormRenderer
-              key={field.id}
-              field={field}
-              path={childPath}
-              arrayChild={type === 'array'}
-              onAddField={onAddField}
-              onFieldUpdate={onFieldUpdate}
-              onRemoveField={onRemoveField}
+      <HCol className="w-full">
+        {child.length > 1 && (
+          <button onClick={() => setCollapsed(!collapsed)} className="-ml-1">
+            <ChevronRightIcon
+              className={twMerge(
+                'ease-in-out duration-300',
+                collapsed ? 'rotate-0' : 'rotate-90',
+              )}
             />
-          ))}
-        </VCol>
-      ) : null}
-      <AddButtons onAddField={onAddField} path={childPath} nested />
+          </button>
+        )}
+        <Input
+          variant="plain"
+          defaultValue={value}
+          placeholder={type === 'array' ? 'Array key' : objectPlaceholder}
+          onBlur={onFieldUpdate(field.id, path)}
+          icon={<XIcon className="text-muted-foreground" />}
+          onIconClick={onRemoveField(field.id, path)}
+          disabled={arrayChild && type === 'object'}
+        />
+      </HCol>
+      <VCol
+        variant="md"
+        className={twMerge(
+          'w-full ease-in-out duration-300',
+          collapsed ? 'max-h-0 opacity-0' : 'max-h-screen opacity-100',
```

Comment:
This `max-h-screen` is good and bad. What if there are a lot of properties, so that the height of these properties is bigger than the users viewport? Some of the inputs will not be visible.  
In this case, I would recommend using hardcoded values such as `max-h-[9999px]`. It can still happen that some inputs are not visible, but for sure it is a higher value than `max-h-screen`. So, even though we are not 100% safe, we are safer. But still, it would be better to calculate the max height through JS.